BUILD_MASTER_GUIDE.md

```diff
@@ -0,0 +1,537 @@
+# Complete Build Guide for HPCC Platform Master Branch
+
+This comprehensive guide provides step-by-step instructions for building the HPCC Platform from the master branch. It consolidates information from various sources and provides modern development workflows.
+
+## Table of Contents
+
+1. [Quick Start](#quick-start)
+2. [Prerequisites](#prerequisites)
+3. [Getting the Source Code](#getting-the-source-code)
+4. [Build Configuration](#build-configuration)
+5. [Building the Platform](#building-the-platform)
+6. [Creating Packages](#creating-packages)
+7. [Testing Your Build](#testing-your-build)
+8. [Development Workflows](#development-workflows)
+9. [Troubleshooting](#troubleshooting)
+
+## Quick Start
+
+### Automated Build Verification
+
+For the fastest way to verify your system and test the build process:
+
+```bash
+# Clone the repository
+git clone --recurse-submodules https://github.com/hpcc-systems/HPCC-Platform.git
+cd HPCC-Platform
+
+# Run the automated verification script
+./verify-build-setup.sh
+```
+
+The verification script will:
+- Check all prerequisites
+- Initialize submodules
+- Configure a test build (client tools only)
+- Verify the build system works
+- Provide next steps
+
+### Manual Quick Start
+
+For experienced developers who want to configure manually:
+
+```bash
+# 1. Install prerequisites (Ubuntu/Debian example)
+sudo apt-get install cmake ninja-build git bison flex build-essential binutils-dev libldap2-dev libcppunit-dev libicu-dev libxslt1-dev zlib1g-dev libboost-regex-dev libarchive-dev python3-dev libv8-dev default-jdk libapr1-dev libaprutil1-dev libiberty-dev libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev nodejs libevent-dev libatlas-base-dev libblas-dev default-libmysqlclient-dev libsqlite3-dev libmemcached-dev libcurl4-openssl-dev pkg-config libtool autotools-dev automake libssl-dev
+
+# 2. Get the source code with submodules
+git clone --recurse-submodules https://github.com/hpcc-systems/HPCC-Platform.git
+cd HPCC-Platform
+
+# 3. Configure the build (first-time setup can take 15-30 minutes)
+mkdir build
+cd build
+cmake -B . -S .. -G Ninja -DCONTAINERIZED=OFF -DUSE_OPTIONAL=OFF -DUSE_CPPUNIT=ON -DINCLUDE_PLUGINS=ON -DSUPPRESS_V8EMBED=ON -DSUPPRESS_REMBED=ON -DCMAKE_BUILD_TYPE=Debug
+
+# 4. Build
+cmake --build . --parallel
+
+# 5. Create package
+cmake --build . --target package
+```
+
+**Note:** The first-time build configuration downloads and compiles many dependencies via vcpkg, which can take 15-30 minutes depending on your system and network connection.
+
+## Prerequisites
+
+The HPCC Platform requires several third-party tools and libraries. The minimum CMake version required is 3.3.2, but 3.15+ is recommended.
```

Comment:
I suspect even more modern is preferred - @GordonSmith which version contains the fix for slow installs?